### PR TITLE
Miscalibration of simulated data

### PR DESCRIPTION
**Pull request by AllaMaevskaya**

### CODE REVIEW

**Comment by AllaMaevskaya** [comment]
Hello
can it be merged?

File: Steer/DigitizerWorkflow/src/SimpleDigitizerWorkflow.cxx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (shahor02) [inline Steer/DigitizerWorkflow/src/SimpleDigitizerWorkflow.cxx]: this triggers an error in fullCI.

File: Detectors/FIT/FT0/workflow/src/ReconstructionSpec.cxx [hunk 2]
```diff
@@ -55,10 +58,10 @@ void ReconstructionDPL::run(ProcessingContext& pc)
   if (mUseMC) {
     LOG(info) << "Ignoring MC info";
   }
-  // auto caliboffsets = pc.inputs().get<o2::ft0::FT0ChannelTimeCalibrationObject*>("ft0offsets");
-  //  mReco.SetChannelOffset(caliboffsets.get());
-  auto caliboffsets = mCCDBManager.get<o2::ft0::FT0ChannelTimeCalibrationObject>("FT0/Calibration/ChannelTimeOffset");
-  mReco.SetChannelOffset(caliboffsets);
+  auto caliboffsets = pc.inputs().get<o2::ft0::FT0ChannelTimeCalibrationObject*>("ft0offsets");
+  mReco.SetChannelOffset(caliboffsets.get());
```
Comment (shahor02) [inline Detectors/FIT/FT0/workflow/src/ReconstructionSpec.cxx]: you don't need to get/set the calibration at every TF, instead, use the mechanism described at https://docs.google.com/presentation/d/1WPNixo57zAn_gM6vMCIBOHs-DZqcKAkLX0CwUgO_VYg/edit#slide=id.g112eef84963_0_3

File: Steer/DigitizerWorkflow/src/FT0DigitizerSpec.cxx [hunk 3]
```diff
@@ -66,7 +73,11 @@ class FT0DPLDigitizerTask : public o2::base::BaseDPLDigitizer
     context->initSimChains(o2::detectors::DetID::FT0, mSimChains);
     const bool withQED = context->isQEDProvided() && !mDisableQED;
     auto& timesview = context->getEventRecords(withQED);
-
+    //set CCDB for miscalibration
+    if (mUseCCDB) {
+      auto caliboffsets = pc.inputs().get<o2::ft0::FT0ChannelTimeCalibrationObject*>("ft0offsets");
```
Comment (shahor02) [inline Steer/DigitizerWorkflow/src/FT0DigitizerSpec.cxx:86]: same here

**Review by shahor02 — CHANGES_REQUESTED** [review]
Hi @AllaMaevskaya 
the fullCI failure is genuine, also, there is more optimal way of using DPL CCDB fetcher, see below.

**Comment by AllaMaevskaya** [comment]
Can it be merge now? I fixed everything you mention

File: Detectors/FIT/FT0/calibration/testWorkflow/LHCphaseCalibrationSpec.h [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef O2_CALIBRATION_LHCCLOCK_CALIBRATOR_H
+#define O2_CALIBRATION_LHCCLOCK_CALIBRATOR_H
```
Comment (shahor02) [inline Detectors/FIT/FT0/calibration/testWorkflow/LHCphaseCalibrationSpec.h:13]: This repeats the guards of Detectors/TOF/calibration/testWorkflow/LHCClockCalibratorSpec.h
The same I see in the Detectors/FIT/FT0/calibration/testWorkflow/GlobalOffsetsCalibrationSpec.h
Could you please change both to FT0-header specific?

**Review by shahor02 — CHANGES_REQUESTED** [review]
Hi @AllaMaevskaya 
There is one more comment, sorry for not noticing before.

**Comment by AllaMaevskaya** [comment]
Is it OK now?

**Comment by shahor02** [comment]
@AllaMaevskaya, looks like your branch was not up to date, you've pushed lot of commits from others.
Could you please rebase it to fresh dev and forcepush again?

Cheers,
 Ruben

**Comment by AllaMaevskaya** [comment]
Done

**Review by shahor02 — APPROVED** [review]
Thanks!

**Comment by shahor02** [comment]
... cannot merge now, github is stuck on `Checking for ability to merge automatically…`
will try tomorrow.

File: Detectors/FIT/FT0/calibration/testWorkflow/LHCphaseCalibrationSpec.h (hunk 4 above)
Comment (AllaMaevskaya) [reply Detectors/FIT/FT0/calibration/testWorkflow/LHCphaseCalibrationSpec.h:13]: Done

**Pull request closed without merging**